tests/conftest.py

```diff
@@ -1,22 +1,44 @@
+# flake8: noqa E402 # See imports after multiprocessing.set_start_method
 import multiprocessing
 import pytest
 import pytest_asyncio
 import tempfile
+
+# Set spawn after stdlib imports, but before other imports
+multiprocessing.set_start_method("spawn")
+
 from pathlib import Path
+from chia.util.keyring_wrapper import KeyringWrapper
+from tests.block_tools import BlockTools, test_constants, create_block_tools
+from tests.util.keyring import TempKeyring
 
 
-multiprocessing.set_start_method("spawn")
+@pytest.fixture(scope="session")
+def get_keychain():
+    with TempKeyring() as keychain:
+        yield keychain
+        KeyringWrapper.cleanup_shared_instance()
+
+
+@pytest.fixture(scope="session", name="bt")
+def bt(get_keychain) -> BlockTools:
+    # Note that this causes a lot of CPU and disk traffic - disk, DB, ports, process creation ...
+    _shared_block_tools = create_block_tools(constants=test_constants, keychain=get_keychain)
+    return _shared_block_tools
+
+
+@pytest.fixture(scope="session")
+def self_hostname(bt):
```

Comment:
oh, I see. You made this a fixture. I think that's a mistake. There's no upside and it adds a lot of extra code.

Reply:
it also means that *any* test that starts a listen port and connects to it is *forced* to create the `BlockTools` fixture. That doesn't make any sense at all.

If you really think it should be a fixture, at least don't make it depend on `bt`. It's just a constant string literal.

Reply:
Well, it's trues that none of the current tests rely on the value of `self_hostname`
It's also true that leaving it the way it is now invades countless test functions with a global.

Reply:
a global *constant*, not variable. it's like naming pi, it won't change, it's a constant.